wordpress-mobile/WordPress-Android
# Stats revamp v2 views and visitors graph

This PR adds new Views and Visitors graph to card in Insights tab on Stats screen.

Fixes #16274 

| Views | Visitors | Marker
|---|---|---|
| ![Screenshot_20220415_130408](https://user-images.githubusercontent.com/990349/163511772-fad50115-83e1-4f4c-8993-4bea8de6f59a.png)|![Screenshot_20220414_230631](https://user-images.githubusercontent.com/990349/163401864-71a5cf25-836f-40f7-8d6a-7bd86f003db4.png)|![Screenshot_20220421_153510](https://user-images.githubusercontent.com/990349/164381687-50f70b78-5535-4367-8a65-101d4e37c9aa.png)|


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab as in the images above (scroll down if necessary)

Test 2:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Notice that the Views graph is in blue and this week and previous week and percent change are all in order
- Notice the graph, data and change are all compare this week with previous week
- Switch to Visitors by tapping chip and notice graph is in purple color

Test 3:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Tap on the graph.  Notice a marker popup as in the third image above.

Test 4:
- Launch Stats management (While on Stats in Insights tab, Click on ⚙️ icon at the top right hand corner)
- Notice there's **Views & Visitors** item under General 
- Tap on **Views & Visitors** item to deselect it and hit SAVE
- Return to Stats and observe the **Views & Visitors** Card is no longer shown
- You may attempt selecting and deselecting a few times to make sure it adds or removes accordingly

Test 5:
- Test with different sites and new sites
- Please check the values are correct particularly

Known Issues:
- The card is added to the end of the list as per existing implementation unless app is a fresh install in which case this card will appear at the top.  Also, when Stats Management is used to add or remove the card gets added to the end of the list
- Clicking on View More takes it to Tags and Categories (temporary setup till a related detail screen is done)
- There are following differences from design.  Need to figure out if possible or not.
  - There are no markers for values on x-axis.  (Need to figure out)
  - Circles appear on all data points when tapped and a MarkerView pops up, not just at the tap point (Need to figure out)
  - Selected Chip color is not dark blue (will be updated later with Styles in a separate PR)

TODO:
- [x] Fix flicker on switching back to Views from Visitors
- [x] Add marker pop-up on graph
- [x] Update accessibility
- [x] Need to add/update tests
- [x] General clean-up and refactoring

## Regression Notes
1. Potential unintended areas of impact
None, it is a new card.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
Updated existing unit test and added new ones.

PR submission checklist:
